### DISCUSSION TITLE
Welcome to Swift Plugin Discussions!

### DISCUSSION THREAD

**Discussion by jansorg** [noctule-ide/noctule-swift-plugin#1] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  I'd appreciate your thoughts, ideas, feature requests and feedback on the Swift plugin!

**Comment by sergeymild** [comment]
how to install it? i cant find it in plugins
<img width="1033" alt="Screenshot 2023-09-23 at 3 42 16 PM" src="https://github.com/jansorg/swift-plugin/assets/3685209/96ec08d6-3794-4a8e-a9a8-db1427b7edea">

**Reply by jansorg** [reply]
It‘s only compatible with 2023.2. Earlier versions are unsupported due to technical limitations. Support for 2023.3 EAP will be published soon.

**Comment by sergeymild** [comment]
i got it, thanks

**Comment by sergeymild** [comment]
On more question) in my toolchain automatically sets xcode, but i dont see any autocompletion or go to declaration or etc it is just has syntax highlight and that is. What do i have to do to have autocompletion? Thanks)
<img width="1120" alt="Screenshot 2023-09-23 at 3 54 41 PM" src="https://github.com/jansorg/swift-plugin/assets/3685209/be5a4a5b-19d9-4ed6-b92d-981f70d311af">

<img width="489" alt="Screenshot 2023-09-23 at 3 56 51 PM" src="https://github.com/jansorg/swift-plugin/assets/3685209/7f2407a0-4d4e-472d-96e3-bc807aa29c07">

**Reply by jansorg** [reply]
The plugin uses Swift's SourceKit LSP server. This server needs index data created by the compiler.
To enable code completion, please run `swift build` (using your configured toolchain) once in your project to create the initial data. Currently, you have to rerun it to keep the data up-to-date, e.g. after refactorings.

**Comment by ronniebarker** [comment]
How do you run `swift build` on an existing project created through xcode (and maintain the xcode compatibility for when the plugin doesn't do something well)?
I have been using AppCode and xCode in combination but want to stop relying on the dead product and am struggling to get going...
